apollo-router/src/axum_factory/axum_http_server_factory.rs

```diff
@@ -402,9 +402,10 @@ where
             (license, Instant::now(), Arc::new(AtomicU64::new(0))),
             license_handler,
         ))
-        .layer(TraceLayer::new_for_http().make_span_with(PropagatingMakeSpan { license }))
         .layer(Extension(service_factory))
-        .layer(cors);
+        .layer(cors)
+        .layer(TraceLayer::new_for_http().make_span_with(PropagatingMakeSpan { license }))
+        .layer(middleware::from_fn(metrics_handler));
```

Comment:
Maybe add a comment for this layer about the importance of its position in this chain of layers.

Reply:
[4d67075](https://github.com/apollographql/router/pull/3919/commits/4d67075301b56ba15ed323014917efbfb4f3ccd4)